e2e/vmss.go

```diff
@@ -24,38 +24,36 @@ const (
 	vmssNameTemplate                         = "abtest%s"
 	listVMSSNetworkInterfaceURLTemplate      = "https://management.azure.com/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/virtualMachineScaleSets/%s/virtualMachines/%d/networkInterfaces?api-version=2018-10-01"
 	loadBalancerBackendAddressPoolIDTemplate = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/aksOutboundBackendPool"
-	maxRetries                               = 3
 )
 
-func bootstrapVMSS(ctx context.Context, t *testing.T, vmssName string, opts *scenarioRunOpts, publicKeyBytes []byte) (*armcompute.VirtualMachineScaleSet, func(), error) {
+func bootstrapVMSS(ctx context.Context, t *testing.T, vmssName string, opts *scenarioRunOpts, publicKeyBytes []byte) (*armcompute.VirtualMachineScaleSet, error) {
 	nodeBootstrapping, err := getNodeBootstrapping(ctx, opts.nbc)
 	if err != nil {
-		return nil, nil, fmt.Errorf("unable to get node bootstrapping: %w", err)
-	}
-
-	cleanupVMSS := func() {
-		log.Printf("deleting vmss %q", vmssName)
-		if _, err := pollVMSSOperation(ctx, vmssName, pollVMSSOperationOpts{
-			pollingInterval: to.Ptr(deleteVMSSPollInterval),
-			pollingTimeout:  to.Ptr(deleteVMSSPollingTimeout),
-		}, func() (Poller[armcompute.VirtualMachineScaleSetsClientDeleteResponse], error) {
-			return config.Azure.VMSS.BeginDelete(ctx, *opts.clusterConfig.cluster.Properties.NodeResourceGroup, vmssName, nil)
-		}); err != nil {
-			t.Errorf("encountered an error while waiting for deletion of vmss %q: %s", vmssName, err)
-		}
-		log.Printf("finished deleting vmss %q", vmssName)
+		return nil, fmt.Errorf("unable to get node bootstrapping: %w", err)
 	}
 
 	vmssModel, err := createVMSSWithPayload(ctx, nodeBootstrapping.CustomData, nodeBootstrapping.CSE, vmssName, publicKeyBytes, opts)
 	if err != nil {
-		return nil, cleanupVMSS, fmt.Errorf("unable to create VMSS with payload: %w", err)
+		return nil, fmt.Errorf("unable to create VMSS with payload: %w", err)
 	}
 
-	return vmssModel, cleanupVMSS, nil
+	if !config.KeepVMSS {
+		t.Cleanup(func() {
+			_, err := config.Azure.VMSS.BeginDelete(ctx, *opts.clusterConfig.Model.Properties.NodeResourceGroup, vmssName, nil)
+			if err != nil {
+				t.Logf("failed to delete vmss %q: %s", vmssName, err)
+			}
+		})
+	}
+
+	return vmssModel, nil
 }
 
 func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName string, publicKeyBytes []byte, opts *scenarioRunOpts) (*armcompute.VirtualMachineScaleSet, error) {
-	model, err := getBaseVMSSModel(vmssName, string(publicKeyBytes), customData, cseCmd, opts)
+	log.Printf("creating VMSS %q in resource group %q", vmssName, *opts.clusterConfig.Model.Properties.NodeResourceGroup)
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
I'd refactor the timeout duration to a separate const

Reply:
Is there a good reason for it?
I personally prefer when single-use magical numbers are inlined. So, when I read the function, I don't need to jump between different places to get the values.

Reply:
fair enough, since I guess we're only using it here